### PR TITLE
PAY-469-updated-types-for-recurring-payments

### PR DESCRIPTION
**Pull request by ShahinMoneyhub**
Changes:
- Updated types for RecurringPaymentRequest

### CODE REVIEW

File: src/schema/payment.ts [hunk 1]
```diff
@@ -25,15 +22,9 @@ export interface AuthRequestPostPayment {
...
 }
 
-type PeriodType =
-  | "Day"
-  | "Week"
-  | "Fortnight"
-  | "Month"
-  | "Half-year"
-  | "Year"
+type PeriodType = "Day" | "Week" | "Fortnight" | "Month" | "Half-year" | "Year";
```
Comment (tomstrat) [inline src/schema/payment.ts:25]: Want to try and make sure your formatter doesn't add semicolons to the end. Its not part of our style. Can see that guide here: https://www.notion.so/moneyhub/Development-process-and-style-guide-081e887009414e88a9fe5e8e3df1bf8a#9febd1cc3e0a4a45ad11c873dd86cc54

File: src/schema/payment.ts [hunk 2]
```diff
@@ -139,27 +127,31 @@ export interface RecurringPaymentSearchParams extends SearchParams {
...
+  | "AcceptedCustomerProfile"
+  | "AcceptedFundsChecked"
+  | "AcceptedWithChange"
+  | "Pending"
   | "Rejected"
-  | "Authorised"
-  | "Revoked"
-  | "Expired"
-
-interface RecurringPayment {
```
Comment (tomstrat) [inline src/schema/payment.ts:149]: Where has this interface gone? Did we need to delete it?

File: src/schema/payment.ts [hunk 3]
```diff
@@ -139,27 +127,31 @@ export interface RecurringPaymentSearchParams extends SearchParams {
 }
 
 type RecurringPaymentStatus =
```
Comment (tomstrat) [inline src/schema/payment.ts:129]: So i think we only return these statuses: https://openbankinguk.github.io/read-write-api-site3/v3.1.11/resources-and-data-models/vrp/domestic-vrps.html#status

The longer list is if we get the `paymentDetails` and i am not too sure if we add that to the normal status. I can have a look and return to this comment

File: src/schema/payment.ts (hunk 1 above)
Comment (ShahinMoneyhub) [reply src/schema/payment.ts]: Removed semicolons now, formatter was to blame

File: src/schema/payment.ts (hunk 2 above)
Comment (ShahinMoneyhub) [reply src/schema/payment.ts:149]: Added back, not sure why I removed it in the first place

File: src/schema/payment.ts (hunk 3 above)
Comment (ShahinMoneyhub) [reply src/schema/payment.ts:139]: Have used the condensed list for now

File: src/schema/payment.ts (hunk 1 above)
Comment (stefaniesatna) [reply src/schema/payment.ts]: It's all reformatted actually, I reckon format on save with prettier installed globally? If so, I think [this](https://www.notion.so/moneyhub/Development-process-and-style-guide-081e887009414e88a9fe5e8e3df1bf8a?pvs=4#9febd1cc3e0a4a45ad11c873dd86cc54) might be useful, though not familiar with the setup in this particular repo

**Review by tomstrat — APPROVED** [review]

**Review by jakefenley-mh — APPROVED** [review]

**Pull request merged**